src/main/java/com/avispl/symphony/dal/infrastructure/management/audinate/dantedirector/DanteDirectorCommunicator.java

```diff
@@ -0,0 +1,874 @@
+	 * The method clears the existing aggregated device list, processes the response, and updates the list accordingly.
+	 * Any error during the process is logged.
+	 */
+	private void populateDeviceDetails() {
+		try {
+			JsonNode response = this.doPost(DanteDirectorConstant.URL, DanteDirectorQuery.DEVICES_INFO, JsonNode.class);
+			if (response.has(DanteDirectorConstant.DATA) && response.get(DanteDirectorConstant.DATA).has(DanteDirectorConstant.DOMAINS)) {
+				cachedData.clear();
+				for (JsonNode domainNode : response.get(DanteDirectorConstant.DATA).get(DanteDirectorConstant.DOMAINS)) {
+					String domainId = domainNode.get(DanteDirectorConstant.ID).asText();
+					if (checkExistDomainId(domainId)) {
+						JsonNode jsonArray = domainNode.get(DanteDirectorConstant.DEVICES);
+						for (JsonNode jsonNode : jsonArray) {
+							JsonNode node = objectMapper.createArrayNode().add(jsonNode);
+
+							String id = jsonNode.get(DanteDirectorConstant.ID).asText();
+							cachedData.removeIf(item -> item.getDeviceId().equals(id));
+							cachedData.addAll(aggregatedDeviceProcessor.extractDevices(node));
+						}
+					}
+				}
+			}
+		} catch (Exception e) {
+			logger.error("Error while populate aggregated device", e);
+		}
+	}
+
+	/**
+	 * Clones and populates a new list of aggregated devices with mapped monitoring properties.
+	 *
+	 * @return A new list of {@link AggregatedDevice} objects with mapped monitoring properties.
+	 */
+	private List<AggregatedDevice> cloneAndPopulateAggregatedDeviceList() {
+		aggregatedDeviceList.clear();
+		synchronized (cachedData) {
+			for (AggregatedDevice item : cachedData) {
+				AggregatedDevice aggregatedDevice = new AggregatedDevice();
+				Map<String, String> cachedValue = item.getProperties();
+				aggregatedDevice.setDeviceId(item.getDeviceId());
+				aggregatedDevice.setDeviceModel(item.getDeviceModel());
+				aggregatedDevice.setDeviceName(item.getDeviceName());
+				aggregatedDevice.setDeviceOnline(item.getDeviceOnline());
+
+				List<AdvancedControllableProperty> controllableProperties = new ArrayList<>();
+				Map<String, String> stats = new HashMap<>();
+				Map<String, String> controlStats = new HashMap<>();
+				mapMonitoringProperty(cachedValue, stats, controlStats, controllableProperties);
+				if (Boolean.TRUE.equals(aggregatedDevice.getDeviceOnline())) {
+					stats.putAll(controlStats);
+					aggregatedDevice.setControllableProperties(controllableProperties);
+				}
+				aggregatedDevice.setProperties(stats);
+				aggregatedDeviceList.add(aggregatedDevice);
+			}
+		}
+		return aggregatedDeviceList;
+	}
+
+	/**
+	 * Checks if a given domain ID exists in the list of domains.
+	 *
+	 * @param id The domain ID to check for existence.
+	 * @return true if the domain ID exists in the list; false otherwise.
+	 */
+	private boolean checkExistDomainId(String id) {
+		for (JsonNode item : domainList) {
+			if (item.get(DanteDirectorConstant.ID).asText().equals(id)) {
```

Comment:
.get() can be null, whereas .at() is at least an empty node. Is it certain that the ID node is always gonna be present?